src/hooks/veNft/VeNftVariants.ts

```diff
@@ -0,0 +1,52 @@
+import axios from 'axios'
+
+import { VeNftVariant } from 'models/v2/veNft'
+import { useQuery } from 'react-query'
+import { ipfsCidUrl } from 'utils/ipfs'
+
+import { VARIANTS_HASH } from 'constants/veNft/veNftProject'
+
+type VeNftMetadataResponse = {
+  metadata: {
+    name: string
+    jbx_range: string
+  }
+}
+
+export function useVeNftVariants() {
+  const hash = VARIANTS_HASH
+  return useQuery(
+    ['nft-variants', hash],
+    async () => {
+      if (!hash) {
+        throw new Error('Variants hash not specified.')
+      }
+      const file = hash + '/characters.json'
+      const url = ipfsCidUrl(file)
+      const response = await axios.get(url)
+      const data: Record<string, VeNftMetadataResponse> = response.data
+      const variants: VeNftVariant[] = Object.entries(data).map(
+        ([id, variant]) => {
+          const { name, jbx_range } = variant.metadata
+          const split = jbx_range.split('-')
+          const tokensStakedMin = parseInt(split[0].replaceAll(',', ''))
```

Comment:
**question** am i understanding this right: we're storing numbers in metadata as `1,000,000` rather than `1000000`?

Might we store an actual `int` instead? Then we don't need to do this parsing.

Reply:
Yes, I will create an issue for this but I am working with a one-time deployment and not totally sure how it happens.

Reply:
Tracked in #1530